### PR TITLE
web/satellite: Payment methods tab

### PR DESCRIPTION
**Pull request by hovex023**
What:  The Payment method section of the new Billing Flow

Why: To create a better end user experience and cleaner flow.

story: https://github.com/storj/storj/issues/4632

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:80]: same. also we can make these methods to be one and get keys as parameter

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:95]: add box-sizing

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:5]: shouldnt this be tr tag and th`s inside?

File: web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue:27]: place svg in svg file and use it like vue component

File: web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue:44]: same

File: web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue:18]: if it is an enum lets use concrete value of it, not magic number

File: web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue:131]: delete comments and extra lines

File: web/satellite/src/components/common/VerticalArrows2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/common/VerticalArrows2.vue:30]: extra

File: web/satellite/src/components/common/VerticalArrows2.vue
Comment (NikolaiYurchenko) [inline web/satellite/src/components/common/VerticalArrows2.vue:16]: extra

**Comment by NikolaiYurchenko** [comment]
also there is posible that we already have this icons. Not 100% sure about this

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:70]: changed

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:80]: Changed it to a switch statement that contains all of the methods.

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:73]: changed

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:59]: changed

File: web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/depositAndBillingHistory/SortingHeader2.vue:5]: Using Divs is following the same design pattern as the other SortingHeader components.

File: web/satellite/src/components/common/VerticalArrows2.vue
Comment (cl-mitch) [reply web/satellite/src/components/common/VerticalArrows2.vue:16]: removed

File: web/satellite/src/components/common/VerticalArrows2.vue
Comment (cl-mitch) [reply web/satellite/src/components/common/VerticalArrows2.vue:30]: removed

File: web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue:131]: removed

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue:207]: Why is flex easier? I would assert that Grid and Flex are equally as useful in css styling. Grid is also used in 'PayoutPeriodCalendar.vue', 'AllSatellitesAuditArea.vue', 'HeldProgress.vue' & 'PayoutPeriodCalendar.vue'

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: removed

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: fixed

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: removed variable. Using 'white' now.

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: fixed

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: fixed

File: web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/CreditCardContainer.vue]: fixed

File: web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue:27]: This is following the original design pattern of the component this was based on.

File: web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/paymentMethods/TokenDepositSelection2.vue:44]: This is following the original design pattern of the component this was based on.

File: web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue
Comment (cl-mitch) [reply web/satellite/src/components/account/billing/paymentMethods/TokenTransactionItem.vue:18]: This is following the original design pattern of the component this was based on.

File: web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue:80]: nit - is this v-if necessary? My understanding is this code is structured like
```
<div v-if="a || b">
    <div v-if="a">...</div>
    <div v-if="b">...</div>
</div>
```
but wouldn't this be sufficient?
```
<div v-if="a">...</div>
<div v-if="b">...</div>
```
mainly I'm just curious. This doesn't block my approval.

File: web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue:144]: nit - I would prefer to use the variable in the message here, `MAX_TOKEN_AMOUNT` rather than hardcoding the amount here

File: web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue:76]: nit - these two constants could be added to src/utils/constants - I don't mind having them defined in the component, but I do think the constants directory is well suited to this purpose

File: web/satellite/src/components/account/billing/paymentMethods/PaymentMethods.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/paymentMethods/PaymentMethods.vue:41]: is this comment still useful? I'm not sure if you added it for your own reference while building this, or if you are adding clarification for future readers. If it's to add clarification, I'm not sure how to interpret the comment

File: web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue:112]: nit - the text styling and top padding of this component looks off to me when testing locally in chrome. Specifically, I notice very small top padding, and a serif font being used for "Edit default card"

File: web/satellite/src/components/account/billing/BillingArea.vue
Comment (mobyvb) [inline web/satellite/src/components/account/billing/BillingArea.vue:6]: you will need to change this flag back before I will approve. Right now, the new billing screen is shown when the flag is turned off
It's fine if you flip it like this to do local testing, but if you commit that change, I can't approve even if everything else looks good. I would suggest flipping the flag back and adding it to a commit. Then, when testing locally just switch it back to `!isNewBillingScreen`, but don't add that change to a future commit. That way, what is pushed on Github always has the expected flag behavior

**Review by mobyvb — COMMENTED** [review]
Overall this change looks good to me. Most of my comments can be ignored, but I'd at least suggest fixing the really easy one (clarifying/removing a comment that was added) if you end up pushing another change anyway.
The only thing blocking my approval is the flag behavior. Once this is fixed, I can approve.

File: web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue:80]: it is necessary because the parent v-if contains 2 different child component inside it that each have their own v-if

File: web/satellite/src/components/account/billing/paymentMethods/PaymentMethods.vue
Comment (hovex023) [reply web/satellite/src/components/account/billing/paymentMethods/PaymentMethods.vue:41]: removed

File: web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue
Comment (mobyvb) [reply web/satellite/src/components/account/billing/billingTabs/PaymentMethods.vue:80]: oh, I didn't notice in my initial review that the parent component was necessary for styling

**Review by mobyvb — APPROVED** [review]

**Pull request closed without merging**
